File: sdi3-180.Cli-SOAP-CSharp/sdi3-180.Cli-SOAP-CSharp/EliminarComentario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_180.Cli_SOAP_CSharp
{
    class EliminarComentario
    {
        private EjbAdminServiceService adminService = new EjbAdminServiceService();
        private rating rating = null;

        public void execute(long id) {
		    comprobarId(id);
		    if (rating == null){ 
			    Console.WriteLine("Comentario inexistente, no puede eliminar");
			    return;
		    }
            adminService.eliminarComentario(id, true);
            Console.WriteLine("Comentario eliminado correctamente");
	    }

        public void comprobarId(long id){
		    foreach(rating r in adminService.getComentarios()){
			    if(id.Equals(r.id))
				    rating=r;
		    }
	    }
    }
}

File: sdi3-180.Cli-SOAP/src/com/sdi/client/accion/EliminarComentario.java
package com.sdi.client.accion;

import com.sdi.ws.AdminService;
import com.sdi.ws.EjbAdminServiceService;
import com.sdi.ws.Rating;

public class EliminarComentario{

	private AdminService adminService = new EjbAdminServiceService().getAdminServicePort();
	private Rating rating = null;
	
	public void execute(Long id) {
		comprobarId(id);
		if (rating == null){ 
			System.err.println("Comentario inexistente, no puede eliminar");
			return;
		}
		adminService.eliminarComentario(id);
		System.out.println("Comentario eliminado correctamente");
	}
	
	public void comprobarId(Long id){
		for(Rating r : adminService.getComentarios()){
			if(id.equals(r.getId()))
				rating=r;
		}
	}

}
